Address safer C++ warnings in RenderLayerCompositor

csaavedra opened the pull request:
#### a4248b6482a3d37d21fe8e3883d2cc6ca9b1a493
<pre>
Address safer C++ warnings in RenderLayerCompositor
<a href="https://bugs.webkit.org/show_bug.cgi?id=288198">https://bugs.webkit.org/show_bug.cgi?id=288198</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::BackingSharingState::backingProviderCandidateForLayer):
(WebCore::RenderLayerCompositor::cacheAcceleratedCompositingFlags):
(WebCore::RenderLayerCompositor::customPositionForVisibleRectComputation const):
(WebCore::RenderLayerCompositor::scheduleRenderingUpdate):
(WebCore::RenderLayerCompositor::visibleRectForLayerFlushing const):
(WebCore::RenderLayerCompositor::flushPendingLayerChanges):
(WebCore::RenderLayerCompositor::didPaintBacking):
(WebCore::RenderLayerCompositor::didChangeVisibleRect):
(WebCore::frameHostingNodeForFrame):
(WebCore::RenderLayerCompositor::updateCompositingLayers):
(WebCore::RenderLayerCompositor::allowBackingStoreDetachingForFixedPosition):
(WebCore::RenderLayerCompositor::collectViewTransitionNewContentLayers):
(WebCore::RenderLayerCompositor::updateBackingAndHierarchy):
(WebCore::RenderLayerCompositor::adjustOverflowScrollbarContainerLayers):
(WebCore::RenderLayerCompositor::layerBecameNonComposited):
(WebCore::RenderLayerCompositor::updateRootContentLayerClipping):
(WebCore::RenderLayerCompositor::updateBacking):
(WebCore::RenderLayerCompositor::repaintOnCompositingChange):
(WebCore::RenderLayerCompositor::repaintInCompositedAncestor):
(WebCore::RenderLayerCompositor::computeExtent const):
(WebCore::RenderLayerCompositor::hasCoordinatedScrolling const):
(WebCore::RenderLayerCompositor::updateScrollLayerPosition):
(WebCore::RenderLayerCompositor::updateScrollLayerClipping):
(WebCore::RenderLayerCompositor::positionForClipLayer const):
(WebCore::RenderLayerCompositor::updateCompositingForLayerTreeAsTextDump):
(WebCore::RenderLayerCompositor::layerTreeAsText):
(WebCore::frameContentsRenderView):
(WebCore::RenderLayerCompositor::attachWidgetContentLayersIfNecessary):
(WebCore::RenderLayerCompositor::setIsInWindow):
(WebCore::RenderLayerCompositor::invalidateEventRegionForAllFrames):
(WebCore::RenderLayerCompositor::updateRootLayerPosition):
(WebCore::isDescendantOfFullScreenLayer):
(WebCore::RenderLayerCompositor::requiresCompositingForTransform const):
(WebCore::RenderLayerCompositor::requiresCompositingForCanvas const):
(WebCore::RenderLayerCompositor::requiresCompositingForScrollableFrame const):
(WebCore::RenderLayerCompositor::fixedLayerIntersectsViewport const):
(WebCore::RenderLayerCompositor::computeCoordinatedPositioningForLayer const):
(WebCore::RenderLayerCompositor::isLayerForIFrameWithScrollCoordinatedContents const):
(WebCore::RenderLayerCompositor::requiresScrollLayer const):
(WebCore::RenderLayerCompositor::paintContents):
(WebCore::RenderLayerCompositor::resetTrackedRepaintRects):
(WebCore::RenderLayerCompositor::isRootFrameCompositor const):
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
(WebCore::RenderLayerCompositor::shouldCompositeOverflowControls const):
(WebCore::RenderLayerCompositor::requiresScrollCornerLayer const):
(WebCore::RenderLayerCompositor::requiresOverhangAreasLayer const):
(WebCore::RenderLayerCompositor::updateLayerForTopOverhangArea):
(WebCore::RenderLayerCompositor::updateLayerForBottomOverhangArea):
(WebCore::RenderLayerCompositor::updateLayerForHeader):
(WebCore::RenderLayerCompositor::updateLayerForFooter):
(WebCore::RenderLayerCompositor::updateLayerForOverhangAreasBackgroundColor):
(WebCore::RenderLayerCompositor::viewHasTransparentBackground const):
(WebCore::RenderLayerCompositor::updateRootContentsLayerBackgroundColor):
(WebCore::RenderLayerCompositor::updateSizeAndPositionForOverhangAreaLayer):
(WebCore::RenderLayerCompositor::updateOverflowControlsLayers):
(WebCore::RenderLayerCompositor::ensureRootLayer):
(WebCore::RenderLayerCompositor::destroyRootLayer):
(WebCore::RenderLayerCompositor::attachRootLayer):
(WebCore::RenderLayerCompositor::detachRootLayer):
(WebCore::RenderLayerCompositor::rootLayerAttachmentChanged):
(WebCore::RenderLayerCompositor::notifyIFramesOfCompositingChange):
(WebCore::RenderLayerCompositor::deviceOrPageScaleFactorChanged):
(WebCore::RenderLayerCompositor::computeFixedViewportConstraints const):
(WebCore::RenderLayerCompositor::computeStickyViewportConstraints const):
(WebCore::RenderLayerCompositor::attachScrollingNode):
(WebCore::RenderLayerCompositor::registerScrollingNodeID):
(WebCore::RenderLayerCompositor::detachScrollCoordinatedLayer):
(WebCore::RenderLayerCompositor::parentRelativeScrollableRect const):
(WebCore::RenderLayerCompositor::updateScrollingNodeLayers):
(WebCore::RenderLayerCompositor::updateScrollingNodeForScrollingRole):
(WebCore::RenderLayerCompositor::updateScrollingNodeForScrollingProxyRole):
(WebCore::RenderLayerCompositor::updateScrollingNodeForPositioningRole):
(WebCore::RenderLayerCompositor::resolveScrollingTreeRelationships):
(WebCore::RenderLayerCompositor::updateSynchronousScrollingNodes):
(WebCore::RenderLayerCompositor::scrollableAreaForScrollingNodeID const):
(WebCore::RenderLayerCompositor::scrollingCoordinator const):
(WebCore::RenderLayerCompositor::protectedPage const):
* Source/WebCore/rendering/RenderLayerCompositor.h:
* Source/WebCore/rendering/RenderWidget.h:
(WebCore::RenderWidget::protectedFrameOwnerElement const):

Canonical link: <a href="https://commits.webkit.org/290869@main">https://commits.webkit.org/290869@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b9df63c0a87b1f19c554c5318fb7d1b2ff27bfa1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10764 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/263 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96223 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41967 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93275 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11162 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19082 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70097 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27610 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94226 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8505 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82645 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50423 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8277 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/328 "Found 3 new test failures: http/tests/security/cookie-module-propagate.html http/tests/security/cookie-module.html imported/w3c/web-platform-tests/cookies/schemeful-same-site/schemeful-websockets.sub.tentative.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41105 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78597 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/234 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98208 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18416 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/13496 "Found 1 new test failure: fast/mediastream/getDisplayMedia-max-constraints5.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79102 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18672 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78476 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78305 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19379 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22814 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/166 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/11590 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18416 "Built successfully") | [💥 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23720 "An unexpected error occured. Recent messages:OS: Sequoia (15.3.1), Xcode: 16.2; Checked out pull request; Found 42407 issues") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18137 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21597 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19910 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

csaavedra commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -1762,7 +1764,7 @@ void RenderLayerCompositor::updateBackingAndHierarchy(RenderLayer& layer, Vector
         // If a negative z-order child is compositing, we get a foreground layer which needs to get parented.
         if (layer.negativeZOrderLayers().size()) {
             if (layerBacking && layerBacking->foregroundLayer())
-                childList.append(*layerBacking->foregroundLayer());
+                childList.append(Ref { *layerBacking->foregroundLayer() });
```
I am not sure why this is needed, @rniwa (`childList` is a reference to a `Vector<Ref<GraphicsLayer>>`): https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/290697@main%20(2085)/scan-build-output/StaticAnalyzer/WebCore/StaticAnalyzerReports/report-RenderLayerCompositor.cpp-updateBackingAndHierarchy-91-02d621.html#EndPath

rniwa replied:
It's because append is considered as a non-trivial function. It's possible we can improve static analyzer to handle this case but it seems okay to add Ref here for now.

csaavedra commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -1798,23 +1800,23 @@ void RenderLayerCompositor::updateBackingAndHierarchy(RenderLayer& layer, Vector
             skipAddToEnclosing = true;
 
         if (!skipAddToEnclosing)
-            childLayersOfEnclosingLayer.append(*layerBacking->childForSuperlayers());
+            childLayersOfEnclosingLayer.append(Ref { *layerBacking->childForSuperlayers() });
```
Same as above.

csaavedra reviewed (commented):
Leaving a couple of comments of changes that I think might not be needed.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -467,18 +467,20 @@ auto RenderLayerCompositor::BackingSharingState::backingProviderCandidateForLaye
-        } else {
-            overlapBounds.setX(clippedOverlapBounds.x());
-            overlapBounds.setWidth(clippedOverlapBounds.width());
+    if (CheckedPtr scrollableArea = providerLayer.scrollableArea()) {
```
We could use: if (~; ~) and not indent the below code block.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -752,8 +754,8 @@ void RenderLayerCompositor::customPositionForVisibleRectComputation(const Graphi
 
-    if (m_renderView.frameView().scrollBehaviorForFixedElements() == ScrollBehaviorForFixedElements::StickToDocumentBounds)
-        scrollPosition = m_renderView.frameView().constrainScrollPositionForOverhang(roundedIntPoint(scrollPosition));
+    if (m_renderView.protectedFrameView()->scrollBehaviorForFixedElements() == ScrollBehaviorForFixedElements::StickToDocumentBounds)
```
Can we store m_renderView.frameView() in a local variable instead of calling m_renderView.protectedFrameView twice?

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -3028,9 +3030,9 @@ auto RenderLayerCompositor::attachWidgetContentLayersIfNecessary(RenderWidget& r
 
         auto* contentsRenderView = frameContentsRenderView(renderer);
-        if (auto frameRootScrollingNodeID = contentsRenderView->frameView().scrollingNodeID()) {
+        if (auto frameRootScrollingNodeID = contentsRenderView->protectedFrameView()->scrollingNodeID()) {
```
Again, can we store contentsRenderView->frameView() in a local variable instead of calling protectedFrameView() twice in succession?

csaavedra replied:
If you mean the call below, notice that it's actually `m_renderView`'s frame view, not `contentsRenderView`.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -3163,18 +3165,18 @@ void RenderLayerCompositor::clearBackingForAllLayers()
-        m_rootContentsLayer->setPosition(m_renderView.frameView().positionForRootContentLayer());
-        m_rootContentsLayer->setAnchorPoint(FloatPoint3D());
+    if (RefPtr rootContentsLayer = m_rootContentsLayer) {
+        rootContentsLayer->setSize(m_renderView.protectedFrameView()->contentsSize());
```
Ditto. We should store m_renderView.frameView() in a local variable instead.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4187,10 +4188,10 @@ bool RenderLayerCompositor::fixedLayerIntersectsViewport(const RenderLayer& laye
     // FIXME: We shouldn't have to check useFixedLayout() here; one of the viewport rects needs to give the correct answer.
     LayoutRect viewBounds;
-    if (m_renderView.frameView().useFixedLayout())
+    if (m_renderView.protectedFrameView()->useFixedLayout())
```
Another place to use a local variable.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4416,16 +4417,16 @@ void RenderLayerCompositor::paintContents(const GraphicsLayer* graphicsLayer, Gr
+        paintScrollbar(RefPtr { m_renderView.frameView().verticalScrollbar() }.get(), context, pixelSnappedRectForIntegralPositionedItems, m_viewBackgroundColor);
     else if (graphicsLayer == layerForScrollCorner()) {
-        const IntRect& scrollCorner = m_renderView.frameView().scrollCornerRect();
+        const IntRect& scrollCorner = m_renderView.protectedFrameView()->scrollCornerRect();
```
Ditto. Store m_renderView.frameView in a local variable instead of calling m_renderView.protectedFrameView twice.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4626,11 +4627,11 @@ GraphicsLayer* RenderLayerCompositor::updateLayerForBottomOverhangArea(bool want
 
-    m_layerForBottomOverhangArea->setPosition(FloatPoint(0, m_rootContentsLayer->size().height() + m_renderView.frameView().headerHeight()
-        + m_renderView.frameView().footerHeight() + m_renderView.frameView().obscuredContentInsets().top()));
+    m_layerForBottomOverhangArea->setPosition(FloatPoint(0, m_rootContentsLayer->size().height() + m_renderView.protectedFrameView()->headerHeight()
```
Here, we're calling m_renderView.protectedFrameView 3 times. We should replace that with a local variable.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4646,24 +4647,24 @@ GraphicsLayer* RenderLayerCompositor::updateLayerForHeader(bool wantsLayer)
 
     m_layerForHeader->setPosition(FloatPoint(0,
-        LocalFrameView::yPositionForHeaderLayer(m_renderView.frameView().scrollPosition(), m_renderView.frameView().obscuredContentInsets().top())));
+        LocalFrameView::yPositionForHeaderLayer(m_renderView.protectedFrameView()->scrollPosition(), m_renderView.protectedFrameView()->obscuredContentInsets().top())));
```
Ditto.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4646,24 +4647,24 @@ GraphicsLayer* RenderLayerCompositor::updateLayerForHeader(bool wantsLayer)
+        LocalFrameView::yPositionForHeaderLayer(m_renderView.protectedFrameView()->scrollPosition(), m_renderView.protectedFrameView()->obscuredContentInsets().top())));
     m_layerForHeader->setAnchorPoint(FloatPoint3D());
-    m_layerForHeader->setSize(FloatSize(m_renderView.frameView().visibleWidth(), m_renderView.frameView().headerHeight()));
+    RefPtr { m_layerForHeader }->setSize(FloatSize(m_renderView.protectedFrameView()->visibleWidth(), m_renderView.protectedFrameView()->headerHeight()));
```
Ditto.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4737,15 +4740,15 @@ bool RenderLayerCompositor::viewNeedsToInvalidateEventRegionOfEnclosingCompositi
 bool RenderLayerCompositor::viewHasTransparentBackground(Color* backgroundColor) const
 {
-    if (m_renderView.frameView().isTransparent()) {
+    if (m_renderView.protectedFrameView()->isTransparent()) {
```
Ditto. Let's not call m_renderView.protectedFrameView 3 times.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4789,12 +4792,12 @@ void RenderLayerCompositor::setRootElementCapturedInViewTransition(bool captured
-        m_rootContentsLayer->setBackgroundColor(m_viewBackgroundColor);
-    else
-        m_rootContentsLayer->setBackgroundColor(Color());
+    if (RefPtr rootContentsLayer = m_rootContentsLayer) {
```
Early exit when m_rootContentsLayer is nullptr looked nicer.

rniwa commented on Source/WebCore/rendering/RenderLayerCompositor.cpp:
```diff
@@ -4838,11 +4841,11 @@ void RenderLayerCompositor::updateSizeAndPositionForOverhangAreaLayer()
 
-    auto obscuredContentInsets = m_renderView.frameView().obscuredContentInsets();
-    IntSize overhangAreaSize = m_renderView.frameView().frameRect().size();
+    auto obscuredContentInsets = m_renderView.protectedFrameView()->obscuredContentInsets();
```
Another place for a local variable.

csaavedra committed:
Address safer C++ warnings in RenderLayerCompositor
https://bugs.webkit.org/show_bug.cgi?id=288198

Reviewed by Ryosuke Niwa.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::BackingSharingState::backingProviderCandidateForLayer):
(WebCore::RenderLayerCompositor::cacheAcceleratedCompositingFlags):
(WebCore::RenderLayerCompositor::customPositionForVisibleRectComputation const):
(WebCore::RenderLayerCompositor::scheduleRenderingUpdate):
(WebCore::RenderLayerCompositor::visibleRectForLayerFlushing const):
(WebCore::RenderLayerCompositor::flushPendingLayerChanges):
(WebCore::RenderLayerCompositor::didPaintBacking):
(WebCore::RenderLayerCompositor::didChangeVisibleRect):
(WebCore::frameHostingNodeForFrame):
(WebCore::RenderLayerCompositor::updateCompositingLayers):
(WebCore::RenderLayerCompositor::allowBackingStoreDetachingForFixedPosition):
(WebCore::RenderLayerCompositor::collectViewTransitionNewContentLayers):
(WebCore::RenderLayerCompositor::updateBackingAndHierarchy):
(WebCore::RenderLayerCompositor::adjustOverflowScrollbarContainerLayers):
(WebCore::RenderLayerCompositor::layerBecameNonComposited):
(WebCore::RenderLayerCompositor::updateRootContentLayerClipping):
(WebCore::RenderLayerCompositor::updateBacking):
(WebCore::RenderLayerCompositor::repaintOnCompositingChange):
(WebCore::RenderLayerCompositor::repaintInCompositedAncestor):
(WebCore::RenderLayerCompositor::computeExtent const):
(WebCore::RenderLayerCompositor::hasCoordinatedScrolling const):
(WebCore::RenderLayerCompositor::updateScrollLayerPosition):
(WebCore::RenderLayerCompositor::updateScrollLayerClipping):
(WebCore::RenderLayerCompositor::positionForClipLayer const):
(WebCore::RenderLayerCompositor::updateCompositingForLayerTreeAsTextDump):
(WebCore::RenderLayerCompositor::layerTreeAsText):
(WebCore::frameContentsRenderView):
(WebCore::RenderLayerCompositor::attachWidgetContentLayersIfNecessary):
(WebCore::RenderLayerCompositor::setIsInWindow):
(WebCore::RenderLayerCompositor::invalidateEventRegionForAllFrames):
(WebCore::RenderLayerCompositor::updateRootLayerPosition):
(WebCore::isDescendantOfFullScreenLayer):
(WebCore::RenderLayerCompositor::requiresCompositingForTransform const):
(WebCore::RenderLayerCompositor::requiresCompositingForCanvas const):
(WebCore::RenderLayerCompositor::requiresCompositingForScrollableFrame const):
(WebCore::RenderLayerCompositor::fixedLayerIntersectsViewport const):
(WebCore::RenderLayerCompositor::computeCoordinatedPositioningForLayer const):
(WebCore::RenderLayerCompositor::isLayerForIFrameWithScrollCoordinatedContents const):
(WebCore::RenderLayerCompositor::requiresScrollLayer const):
(WebCore::RenderLayerCompositor::paintContents):
(WebCore::RenderLayerCompositor::resetTrackedRepaintRects):
(WebCore::RenderLayerCompositor::isRootFrameCompositor const):
(WebCore::RenderLayerCompositor::isMainFrameCompositor const):
(WebCore::RenderLayerCompositor::shouldCompositeOverflowControls const):
(WebCore::RenderLayerCompositor::requiresScrollCornerLayer const):
(WebCore::RenderLayerCompositor::requiresOverhangAreasLayer const):
(WebCore::RenderLayerCompositor::updateLayerForTopOverhangArea):
(WebCore::RenderLayerCompositor::updateLayerForBottomOverhangArea):
(WebCore::RenderLayerCompositor::updateLayerForHeader):
(WebCore::RenderLayerCompositor::updateLayerForFooter):
(WebCore::RenderLayerCompositor::updateLayerForOverhangAreasBackgroundColor):
(WebCore::RenderLayerCompositor::viewHasTransparentBackground const):
(WebCore::RenderLayerCompositor::updateRootContentsLayerBackgroundColor):
(WebCore::RenderLayerCompositor::updateSizeAndPositionForOverhangAreaLayer):
(WebCore::RenderLayerCompositor::updateOverflowControlsLayers):
(WebCore::RenderLayerCompositor::ensureRootLayer):
(WebCore::RenderLayerCompositor::destroyRootLayer):
(WebCore::RenderLayerCompositor::attachRootLayer):
(WebCore::RenderLayerCompositor::detachRootLayer):
(WebCore::RenderLayerCompositor::rootLayerAttachmentChanged):
(WebCore::RenderLayerCompositor::notifyIFramesOfCompositingChange):
(WebCore::RenderLayerCompositor::deviceOrPageScaleFactorChanged):
(WebCore::RenderLayerCompositor::computeFixedViewportConstraints const):
(WebCore::RenderLayerCompositor::computeStickyViewportConstraints const):
(WebCore::RenderLayerCompositor::attachScrollingNode):
(WebCore::RenderLayerCompositor::registerScrollingNodeID):
(WebCore::RenderLayerCompositor::detachScrollCoordinatedLayer):
(WebCore::RenderLayerCompositor::parentRelativeScrollableRect const):
(WebCore::RenderLayerCompositor::updateScrollingNodeLayers):
(WebCore::RenderLayerCompositor::updateScrollingNodeForScrollingRole):
(WebCore::RenderLayerCompositor::updateScrollingNodeForScrollingProxyRole):
(WebCore::RenderLayerCompositor::updateScrollingNodeForPositioningRole):
(WebCore::RenderLayerCompositor::resolveScrollingTreeRelationships):
(WebCore::RenderLayerCompositor::updateSynchronousScrollingNodes):
(WebCore::RenderLayerCompositor::scrollableAreaForScrollingNodeID const):
(WebCore::RenderLayerCompositor::scrollingCoordinator const):
(WebCore::RenderLayerCompositor::protectedPage const):
* Source/WebCore/rendering/RenderLayerCompositor.h:
* Source/WebCore/rendering/RenderWidget.h:
(WebCore::RenderWidget::protectedFrameOwnerElement const):

Canonical link: https://commits.webkit.org/290869@main
